core/src/main/java/io/fabric8/maven/core/service/kubernetes/JibBuildServiceUtil.java

```diff
@@ -0,0 +1,192 @@
+import io.fabric8.maven.docker.util.MojoParameters;
+import org.apache.maven.plugin.MojoExecutionException;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class JibBuildServiceUtil {
+
+    private JibBuildServiceUtil() {}
+
+    private static final String DEFAULT_JAR_NAME = "/app.jar";
+
+    public static void buildImage(JibBuildConfiguration buildConfiguration, Logger log) throws InvalidImageReferenceException {
+
+        String fromImage = buildConfiguration.getFrom();
+        String targetImage = buildConfiguration.getTargetImage();
+        Credential credential = buildConfiguration.getCredential();
+        Map<String, String> envMap  = buildConfiguration.getEnvMap();
+        List<String> portList = buildConfiguration.getPorts();
+        Set<Port> portSet = getPortSet(portList);
+        String  outputDir = buildConfiguration.getOutputDir();
+        String targetDir = buildConfiguration.getTargetDir();
+        Path fatJar = buildConfiguration.getFatJar();
+
+        List<String> entrypointList = new ArrayList<>();
+        if(buildConfiguration.getEntryPoint() != null) {
+            entrypointList = buildConfiguration.getEntryPoint().asStrings();
+        }
+
+        buildImage(fromImage, targetImage, envMap, credential, portSet, fatJar, entrypointList, targetDir, outputDir, log);
+    }
+
+    private static Set<Port> getPortSet(List<String> ports) {
+
+        Set<Port> portSet = new HashSet<Port>();
+        for(String port : ports) {
+            portSet.add(Port.tcp(Integer.parseInt(port)));
+        }
+
+        return portSet;
+    }
+
+    protected static JibContainer buildImage(String baseImage, String targetImage, Map<String, String> envMap, Credential credential, Set<Port> portSet, Path fatJar, List<String> entrypointList, String targetDir, String outputDir, Logger log) throws InvalidImageReferenceException {
+
+        String username = "";
+        String password = "";
+        String imageTarName = ImageReference.parse(targetImage).getRepository().concat(".tar");
+
+        JibContainerBuilder contBuild = Jib.from(baseImage);
+
+        if (envMap != null) {
+            contBuild = contBuild.setEnvironment(envMap);
+        }
+
+        if (portSet != null) {
+            contBuild = contBuild.setExposedPorts(portSet);
+        }
+
+
+        if (fatJar != null) {
+            String fatJarName = fatJar.getFileName().toString();
+            String jarPath = targetDir + "/" + (fatJarName.isEmpty() ? DEFAULT_JAR_NAME: fatJarName);
+            contBuild = contBuild
+                    .addLayer(LayerConfiguration.builder().addEntry(fatJar, AbsoluteUnixPath.get(jarPath)).build());
+        }
+
+        if(!entrypointList.isEmpty()) {
+            contBuild = contBuild.setEntrypoint(entrypointList);
+        }
+
+        if (credential != null) {
```

Comment:
what happens when `credential` is null? I see username and password are crucial dependencies for image generation.

Reply:
A tar image is generated locally